journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -165,7 +165,6 @@ public void deleteUntil(final long index) {
       compactSegments.clear();
 
       journalIndex.deleteUntil(index);
-      resetHead(getFirstSegment().index());
```

Comment:
When deleting, segments are only marked for deletion. They will be deleted only when there are no more readers on that segment. The readers on the deleted segment will not read anything until it is reset to a valid index. This is not done automatically anymore. This should be done explicitly. For example, here when we compact we know that there won't be any readers on that segment. Otherwise we will never compact it. But journal doesn't care about it.

Reply:
Ok thanks sounds good. I probably need to see the implementation to understand better how do we achieve this :)